README.md

```diff
@@ -162,6 +162,9 @@ In this case, Juttle will issue a query against ES that matches documents whose
 * `reduce by fieldname` (other than reduce by document type)
 * `reduce -every :interval:`
 
+##### Optimization and nested objects
+There are a few fundamental incompatibilities between [Elasticsearch's model](https://www.elastic.co/guide/en/elasticsearch/guide/current/complex-core-fields.html) for nested object and array fields and [Juttle's](http://juttle.github.io/juttle/concepts/fields/#fields-with-object-or-array-values). This can lead to some odd results for optimized programs. For objects, an optimized `reduce by some_object_field` will return `null` as the only value for `some_object_field`. For arrays, an optimized `reduce by some_array_field` will return a separate value for `some_array_field` for every element in every array stored in `some_array_field`. 
+
```

Comment:
Advice for how to get better behavior would be fitting here. Perhaps:
```
For results conforming to Juttle's `reduce` behavior, if reducing by an object or array field, make the reduce follow unoptimized path by adding `put unoptimized = true` before the `reduce by some_object_or_array_field`.
```
(My read of the situation is that we don't have a way to force the program to do that, because by the time we discover that way an object field, it's too late; hence, the user has to do that.)